src/libraries/System.Private.CoreLib/src/System/Globalization/InvariantModeCasing.cs

```diff
@@ -215,7 +215,7 @@ internal static int CompareStringIgnoreCase(ref char strA, int lengthA, ref char
                     continue;
                 }
 
-                return (int)codePointA - (int)codePointB;
+                return (int)aUpper - (int)bUpper;
```

Comment:
I'm surprised we didn't have cases where this returned the wrong result entirely, for instance when comparing a single upper case to lower case.  Do we need more test cases?

Reply:
This case scoped to the globalization invariant mode. The test I added here should be good enough to catch the cases we need to fix. 

Note, we have the right implementation in the other places, like 

https://github.com/dotnet/runtime/blob/46c9a4fff83f35ec659e6659050440aadccf3201/src/libraries/System.Private.CoreLib/src/System/Globalization/Ordinal.cs#L52

and 

https://github.com/dotnet/runtime/blob/46c9a4fff83f35ec659e6659050440aadccf3201/src/libraries/System.Private.CoreLib/src/System/Globalization/OrdinalCasing.Icu.cs#L231

Reply:
I was just imagining that it might give the wrong answers (let alone ordering) for things like:
`a` `B` -> 97 - 65 = 32 **<-- incorrect**
`A` `b` -> 65 - 98 = -33
`A` `B` -> 65 - 66 = -1

And I was surprised we didn't have a theory test somewhere for such basic comparisons.

Reply:
Yes, I understand what you are saying but does it really matter much testing single character string or a string with multiple characters as the case I added? if you think this is worth adding I can do it.